pkg/agent/agent.go

```diff
@@ -142,8 +105,7 @@ func (k *Klocksmith) Run() error {
 	}
 
 	// watch update engine for status updates
-	watchUpdateStatusStop := make(chan struct{})
-	go k.watchUpdateStatus(k.updateStatusCallback, watchUpdateStatusStop)
+	go k.watchUpdateStatus(k.updateStatusCallback, stop)
```

Comment:
Using `stop` (instead of a new channel) means all goroutines spawned fanning out from here can pass the same channel so that all watches can be stopped from the top-level stop channel.

It'd be slighly nicer to have a `merge(stop <-chan struct{}...) stop <-chan struct{}` that was stopped by any of the input channels being stopped since for this particular goroutine we don't care about it after line 160.